### DISCUSSION TITLE
How do I convert a Image to a byte Array?

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [SixLabors/ImageSharp#2395] [Q&A]
CopyPixelDataTo doesn't exist and there is no suggestion in vscode on the namespace it's in if it still exists.
```cs
using OpenTK.Mathematics;
using OpenTK.Graphics.ES30;
using SixLabors.ImageSharp;
using SixLabors.ImageSharp.Advanced;
using SixLabors.ImageSharp.Processing;
using SixLabors.ImageSharp.PixelFormats;
```
```cs
    public Texture(string file) : this()
    {
        if (!File.Exists(file)) return;
        
        using (var image = Image.Load(file))
        {
            image.Mutate(x=>x.Flip(FlipMode.Vertical));
            image.CopyPixelDataTo(this.data);
        }
        Init();
    }
```

**Comment by tiesont** [comment]
[Save](https://github.com/SixLabors/ImageSharp/blob/main/src/ImageSharp/Image.cs#L118) (or the async version) can be used to save the image to a stream - save to a MemoryStream and then call [ToArray()](https://learn.microsoft.com/en-us/dotnet/api/system.io.memorystream.toarray?view=net-7.0#system-io-memorystream-toarray)?

**Reply by antonfirsov** [reply]
That would contain an image stream encoded in a specific image format. `CopyPixelDataTo` writes raw pixel data of the original pixel type, which is a different use-case.

**Comment by JimBobSquarePants** [comment]
The docs are clear here. The method exists on the generic `Image<T>` type.

https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Image-1.html#SixLabors_ImageSharp_Image_1_CopyPixelDataTo_

**Reply by antonfirsov** [reply]
And the reason for that is that you need to know your pixel type when working with raw pixel data @Shadowblitz16. Here is an OpenTK tutorial with ImageSharp: https://opentk.net/learn/chapter1/5-textures.html#imagesharp
